common/headtracker/types/head_saver.go

```diff
@@ -0,0 +1,21 @@
+package types
+
+import (
+	"context"
+
+	commontypes "github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// HeadSaver is an chain agnostic interface for saving and loading heads
+// Each chain will have its own custom implementation of HeadSaver interface.
```

Comment:
After our latest discussion, HeadSaver will just be implemented once right?
Different chains will just instantiate this generic type with their native Head and BlockHash types.

Reply:
yup, sorry for the lack of clarity. I meant that every chain will implement the interface as struct.  Will update the comments